File: src/main/java/mini/project/daaw/Model/Module.java
package mini.project.daaw.Model;

public class Module {
    private String moduleName;
    private String week;
    private String type;
    private String file;
    private String pdfName;
    private String period;

    public Module(String moduleName, String week, String type, String file, String pdfName, String period) {
        this.moduleName = moduleName;
        this.week = week;
        this.type = type;
        this.file = file;
        this.pdfName = pdfName;
        this.period = period;
    }

    public Module() {
    }

    public String getModuleName() {
        return moduleName;
    }

    public String getWeek() {
        return week;
    }

    public String getType() {
        return type;
    }

    public String getFile() {
        return file;
    }

    public String getPdfName() {
        return pdfName;
    }

    public String getPeriod() {return period; }

    ///setters

    public void setModuleName(String moduleName) {
        this.moduleName = moduleName;
    }

    public void setWeek(String week) {
        this.week = week;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setFile(String file) {
        this.file = file;
    }

    public void setPdfName(String pdfName) {
        this.pdfName = pdfName;
    }

    public void setPeriod(String period) {
        this.period = period;
    }
}

File: src/main/webapp/WEB-INF/jsfiles/module.js
const editBtns = document.querySelectorAll("tbody tr .modifier");
let data = [
    <c:forEach items="${ info }" var="info">
        {
            moduleName: "<c:out value="${info.getModuleName()}"/>",
            week: "<c:out value="${info.getWeek()}"/>",
            type: "<c:out value="${info.getType()}"/>",
            file: "<c:out value="${info.getFile()}"/>",
            pdfName: "<c:out value="${info.getPdfName()}"/>",
            period: "<c:out value="${info.getPeriod()}"/>"
        },
    </c:forEach>
]

editBtns.forEach((editBtn, index) => {
    editBtn.addEventListener("click", () => {
        console.log(data[index])
        editInputs(data[index].week, data[index].type, data[index].file, data[index].pdfName, data[index].period);
    })
})

//////
const week = document.querySelector(".inp.week");
const type = document.querySelector(".typek");
const file = document.querySelector(".inp.file");
const pdfName = document.querySelector(".inp.pdfName");
const period = document.querySelector(".inp.period");


let editInputs = (str1, str2, str3, str4, str5) => {
    week.value = str1;
    type.value = str2;
    file.value = str3;
    pdfName.value = str4;
    period.value = str5;
    week.readOnly = true;
    pdfName.readOnly = true;
    week.style.opacity = 0.6;
    type.style.opacity = 0.6;
    pdfName.style.opacity = 0.6;
}






